### PR TITLE
[New] Add kindeditor@4.1.11 w/ git auto-update

### PR DESCRIPTION
**Pull request by maruilian11**
**git repo url:** https://github.com/kindsoft/kindeditor
**Watch 100
  Star 725
 Fork 316**
@PeterDaveHello  could u help me check this PR for #6153 ? thank you!
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [ ] No already exist / duplicated issue and PR 
- [x] The lib has notable popularity
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry
- [x] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update) https://github.com/kindsoft/kindeditor/issues/196
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
Don't we need `lang/`?

**Comment by maruilian11** [comment]
@Amomo i added `lang` `plugins` and `themes`, could you help me check this PR? thanks!

**Comment by PeterDaveHello** [comment]
You can not partially minify the lib, our bot doesn't support that.

**Comment by PeterDaveHello** [comment]
v4.1.10 in title, v4.1.11 in commit?

**Comment by Amomo** [comment]
Hi @maruilian11 sorry for late.
I think this PR is ok, could you rebase it with latest master branch? Thank you!

**Comment by maruilian11** [comment]
@Amomo i rebased the branch with our latest master branch, could you help me review this PR again? thanks!

**Comment by PeterDaveHello** [comment]
There is only v4.1.10 on npm, so how do you get v4.1.11?

**Comment by maruilian11** [comment]
@PeterDaveHello sorry for not mentioning that, I get these files from [git tag v4.1.11 zip](https://github.com/kindsoft/kindeditor/archive/v4.1.11.zip)

**Comment by PeterDaveHello** [comment]
@maruilian11 you are doing somthing tricky, think about it, this PR is not acceptable.

**Comment by maruilian11** [comment]
@PeterDaveHello I just found out there is only one version on npm package, and it seems not  official...
so I change to use git auto-update config.
dose this commit still have something too tricky?

**Comment by PeterDaveHello** [comment]
We discussed about the same problem before. The bot doesn't support partially minify certain directories, you can not do that.

**Comment by maruilian11** [comment]
@Piicksarn according to http://kindeditor.org/docs/install and http://kindeditor.net/docs/usage.html,
I chose to add `/lang` and `/theme`, and minified all `.css` and `.js` in the two folder.
could you help me check the PR? thanks!

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by maruilian11** [comment]
@cdnjs/intern2 Please help me review this PR, thanks!

**Comment by maruilian11** [comment]
@PeterDaveHello Since we can't partially minify the certain folder, do we just not provide the minified files from our side in this case.

**Comment by PeterDaveHello** [comment]
have you ask the authors about if they can provide it?

**Comment by maruilian11** [comment]
@PeterDaveHello Still get no response from the issue, do we add this lib without auto-update first? Maybe manually add it first before we can solve the partially minified certain directory?

**Comment by PeterDaveHello** [comment]
Okay let's do it.

**Comment by sufuf3** [comment]
@maruilian11 how about email(luolonghao@gmail.com) to the author to ask for the issue?
I found the author's email from https://github.com/luolonghao
Thank you.

**Comment by maruilian11** [comment]
@sufuf3 If the author doesn't use GitHub for a long time, it may mean that the author don't want to maintain the lib anymore, or just don't have time now. Then, I prefer don't bother the author with the email. Of course, if Peter ask me to do, I'll email the author. If you would love to email for this, I'll wait for your response. Thanks.

**Comment by PeterDaveHello** [comment]
@maruilian11 why not? Let's just give it a try.

**Comment by maruilian11** [comment]
E-mail Sent.

**Comment by maruilian11** [comment]
@PeterDaveHello Someone sent another PR to add this lib, and I still don't get any response from the author. Do we continue this PR and close https://github.com/cdnjs/cdnjs/pull/11867?

**Comment by huaiping** [comment]
KindEditor doesn't work without the folder "plugins", so i sent a new PR @maruilian11 @PeterDaveHello

**Comment by PeterDaveHello** [comment]
@huaiping  we prefer to continue the existing PR, so the new one may not be accepted and merged, still thanks for the updates.

**Comment by PeterDaveHello** [comment]
@maruilian11 @sufuf3 what's the current status here? Can we manually get it done ASAP?

File: ajax/libs/kindeditor/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "homepage": "http://kindeditor.org",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/kindsoft/kindeditor.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "kindeditor*",
+          "+(lang|themes)/**/*.+(js|css|gif|png)"
```
Comment (maruilian11) [inline ajax/libs/kindeditor/package.json]: @PeterDaveHello I remember that, we can't set partially minify action in one certain directory. But can we only set minify setting for `lang` and `themes`?

Comment (sufuf3) [reply ajax/libs/kindeditor/package.json]: I think it's fine.
There are some cases in CDNJS.
cc @extend1994 @PeterDaveHello

Comment (PeterDaveHello) [reply ajax/libs/kindeditor/package.json]: We can't, what's the case? @sufuf3

Comment (sufuf3) [reply ajax/libs/kindeditor/package.json]: I answer the question `But can we only set minified setting for lang and themes?`
Sorry not mention detail.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer Just update the commit to add `plugins` folder. Thank for huaiping's reminding.

**Comment by sufuf3** [comment]
@extend1994 please help review this PR, thanks.

File: ajax/libs/kindeditor/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  "homepage": "http://kindeditor.org",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/kindsoft/kindeditor.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "kindeditor*",
+          "+(lang|themes|plugins)/**/!(prettify)*.+(js|css|gif|png)"
```
Comment (extend1994) [inline ajax/libs/kindeditor/package.json]: Should also add .map postfix?
I can't find the same `prettify.css` & `prettify.js` on cdnjs or https://github.com/google/code-prettify. Not sure why it's excluded?

**Comment by sashberd** [comment]
@PeterDaveHello @sufuf3 @extend1994 Cannot rebase this PR, get auth error. Please help me to rebase it or close it.

**Comment by PeterDaveHello** [comment]
@sashberd not sure if you realize that yet, just a small reminder here.

For those old PRs before GitHub has the feature to help maintainers revise, the contributor who sends the PRs need to check the checkbox "Allow edits from maintainers. "

There is information that can help you know if you can do that, if GitHub displays this message at the end of the PR discussion: "Add more commits by pushing to the PR-Branch-Name branch on CONTRIBUTOR/cdnjs.", it means you can help modify it, otherwise, you cannot.

You can also see the info from api, e.g.,
https://api.github.com/repos/cdnjs/cdnjs/pulls/8146

The value of "maintainer_can_modify" which can be true or false which means you can or can not help revise it.

Hope this tip can help you save some time.

**Comment by sashberd** [comment]
@PeterDaveHello Thanks for the tip

**Comment by extend1994** [comment]
@maruilian11 We will need to close this PR if there is no response in next 7 days, thanks.

**Comment by extend1994** [comment]
Close this PR due to no response, would be great if anyone interested in it help with this PR or related issue.

**Pull request closed without merging**